Code/Framework/AzFramework/AzFramework/Quality/QualityCVarGroup.cpp

```diff
@@ -0,0 +1,324 @@
+        }
+
+        return QualityLevel(intValue);
+    }
+
+    QualityLevel QualityCVarGroup::FromEnum(const AZ::CVarFixedString& value) const
+    {
+        using EnumMemberPair = typename AZ::AzEnumTraits<QualityLevel>::MembersArrayType::value_type;
+        auto findEnumOptionValue = [&value](EnumMemberPair enumPair)
+        {
+            // case-insensitive comparison
+            constexpr bool caseSensitive{ false };
+            return AZ::StringFunc::Equal(value, enumPair.m_string, caseSensitive);
+        };
+        if (auto foundIt = AZStd::ranges::find_if(AZ::AzEnumTraits<QualityLevel>::Members,
+            findEnumOptionValue); foundIt != AZ::AzEnumTraits<QualityLevel>::Members.end())
+        {
+            return foundIt->m_value;
+        }
+
+        return QualityLevel::Invalid;
+    }
+
+    QualityLevel QualityCVarGroup::FromName(const AZ::CVarFixedString& value) const
+    {
+        using FixedValueString = AZ::SettingsRegistryInterface::FixedValueString;
+        using VisitResponse = AZ::SettingsRegistryInterface::VisitResponse;
+
+        auto registry = AZ::SettingsRegistry::Get();
+        if (registry == nullptr)
+        {
+            return QualityLevel::Invalid;
+        }
+
+        QualityLevel qualityLevel{ QualityLevel::Invalid };
+
+        // visit each user-defined level name and perform a case-insensitive comparison
+        int32_t arrayIndex{ 0 };
+        auto callback = [&](const auto& visitArrayArg)
+        {
+            FixedValueString qualityLevelName;
+            if (registry->Get(qualityLevelName, visitArrayArg.m_jsonKeyPath))
+            {
+                // case-insensitive comparison
+                constexpr bool caseSensitive{ false };
+                if (AZ::StringFunc::Equal(qualityLevelName, value, caseSensitive))
+                {
+                    qualityLevel = QualityLevel(arrayIndex);
+                    return VisitResponse::Done;
+                }
+            }
+            arrayIndex++;
+            return VisitResponse::Continue;
+        };
+        AZ::SettingsRegistryVisitorUtils::VisitArray(*registry, callback,
+            FixedValueString::format("%s/Levels", m_path.c_str()));
+
+        return qualityLevel;
+    }
+
+    void QualityCVarGroup::LoadQualityLevel(QualityLevel qualityLevel)
+    {
+        using FixedValueString = AZ::SettingsRegistryInterface::FixedValueString;
+        using VisitResponse = AZ::SettingsRegistryInterface::VisitResponse;
+        using Type = AZ::SettingsRegistryInterface::Type;
+
+        auto registry = AZ::SettingsRegistry::Get();
+        if (registry == nullptr)
+        {
+            return;
+        }
+
+        auto callback = [&](const auto& visitArgs)
+        {
+            AZStd::string_view command = visitArgs.m_fieldName;
+
+            if (visitArgs.m_type == Type::Object ||
+                visitArgs.m_type == Type::Null ||
+                visitArgs.m_type == Type::NoType)

```

Comment:
nit: Maybe a `switch` construct on `visitArgs.m_type` can make this block easier to read